### DISCUSSION TITLE
Push Option

### DISCUSSION THREAD

**Discussion by ghost** [anasty17/mirror-leech-telegram-bot#1179] [Q&A] [answered]
When we hit push button, all changed code will applied or still need restart?

**Comment by anasty17** [comment, marked as answer]
I didn't add details yet for this some need restart and some not!

Config.env need restart if u didn't add database and if u added database then only values that u can't change from bot settings will get updated. But if u have UPSTREAM_REPO u must pushed because after restart u will get files from UPSTREAM_REPO.

I will add load function for Config.env when upload new one.

List_drives.txt no need to restart even if u didn't pushed . But it will not get saved if u didn't add database but if u pushed it then next restart u will get them.

Account.zip restart not needed.

Token.pickle restart not needed.

Netrc restart not needed but there a bug regarding aria2c I'm fixing it.

Cookies.txt no restart needed.

**Reply by ghost** [reply]
> I didn't add details yet for this some need restart and some not!
> 
> Config.env need restart if u didn't add database and if u added database then only values that u can't change from bot settings will get updated. But if u have UPSTREAM_REPO u must pushed because after restart u will get files from UPSTREAM_REPO.
> 
> I will add load function for Config.env when upload new one.
> 
> List_drives.txt no need to restart even if u didn't pushed . But it will not get saved if u didn't add database but if u pushed it then next restart u will get them.
> 
> Account.zip restart not needed.
> 
> Token.pickle restart not needed.
> 
> Netrc restart not needed but there a bug regarding aria2c I'm fixing it.
> 
> Cookies.txt no restart needed.

Okey sir 👍👍

**Reply by anasty17** [reply]
It's changed now but not tested well
Config will load when u upload it but not everything

**Reply by ghost** [reply]
> It's changed now but not tested well Config will load when u upload it but not everything

Some import still missing sir

**Reply by anasty17** [reply]
Honestly i didn't test load_config function

**Reply by anasty17** [reply]
Now try forget to import environ. Maybe there are more issues idk

**Reply by ghost** [reply]
> Now try forget to import environ. Maybe there are more issues idk

Also user_data sir
